packages/aws-cdk-lib/aws-servicecatalog/lib/private/product-stack-synthesizer.ts

```diff
@@ -30,7 +30,7 @@ export class ProductStackSynthesizer extends cdk.StackSynthesizer {
         cdk.Annotations.of(parentStack).addWarning('[WARNING] Bucket Policy Permissions cannot be added to' +
           ' referenced Bucket. Please make sure your bucket has the correct permissions');
       }
-      this.bucketDeployment = new BucketDeployment(parentStack, 'AssetsBucketDeployment', {
+      this.bucketDeployment = new BucketDeployment(parentStack, `${cdk.Names.uniqueId(this.boundStack)}-AssetsBucketDeployment`, {
```

Comment:
Since the bucket deployment can be shared within the same stack what if we did something like this:

```ts
this.bucketDeployment = parentStack.tryFindChild('AssetsBucketDeployment') ?? new BucketDeployment(...)
```